sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -154,24 +174,175 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
             .select("COUNT(*)")
             .from(EducationProgramOffering, "offerings")
             .where("offerings.educationProgram.id = programs.id")
+
+            .andWhere("offerings.offeringType in (:...offeringTypes)", {
+              offeringTypes,
+            })
             .andWhere("offerings.institutionLocation.id = :locationId", {
               locationId,
             }),
         "totalofferings",
       )
-      .where("programs.institution.id = :institutionId", { institutionId })
-      .getRawMany();
+      .where("programs.institution.id = :institutionId", { institutionId });
+
+    const queryParams: any[] = [...offeringTypes, locationId, institutionId];
+    // program name search
+    if (paginationOptions.searchName) {
+      summaryResult.andWhere("programs.name Ilike :searchProgramName", {
+        searchProgramName: `%${paginationOptions.searchName}%`,
+      });
+      queryParams.push(`%${paginationOptions.searchName}%`);
+    }
+
+    // total raw count
+    const totalCount = await getRawCount(
+      this.repo,
+      summaryResult.getSql(),
+      queryParams,
+    );
+
+    // pagination
+    summaryResult
+      .take(paginationOptions.pageLimit)
+      .skip(paginationOptions.page * paginationOptions.pageLimit);
+
+    // sort
+    if (paginationOptions.sortField && paginationOptions.sortOrder) {
+      summaryResult.orderBy(
+        databaseFieldOfInstitutionProgramDataTable(paginationOptions.sortField),
+        paginationOptions.sortOrder,
+      );
+    } else {
+      // default sort and order
+      summaryResult.orderBy(
+        databaseFieldOfInstitutionProgramDataTable(DEFAULT_SORT_FIELD),
+        FieldSortOrder.ASC,
+      );
+    }
+
+    const programs = await summaryResult.getRawMany();
 
-    return summaryResult.map((summary) => {
+    const programSummary = programs.map((summary) => {
       const summaryItem = new EducationProgramsSummary();
       summaryItem.id = summary.id;
       summaryItem.name = summary.name;
       summaryItem.cipCode = summary.cipcode;
       summaryItem.credentialType = summary.credentialtype;
+      summaryItem.credentialTypeToDisplay = credentialTypeToDisplay(
+        summary.credentialtype,
+      );
       summaryItem.approvalStatus = summary.approvalstatus;
       summaryItem.totalOfferings = summary.totalofferings;
       return summaryItem;
     });
+    return {
+      programsSummary: programSummary,
+      totalProgram: totalCount[0].count,
```

Comment:
Is there a way that we can get const[totalCount] =  await getRawCount() so that you can skip the index reference.

Reply:
this code was changed